mesonbuild/dependencies/base.py

```diff
@@ -1477,10 +1480,23 @@ def _detect_dep(self, name: str, modules: T.List[T.Tuple[str, bool]], components
                     otherDeps += [x for x in tgt.properties['IMPORTED_LINK_DEPENDENT_LIBRARIES'] if x]
 
                 for j in otherDeps:
+                    is_abs_path = os.path.split(j)[0]
                     if j in self.traceparser.targets:
                         targets += [j]
-                    elif reg_is_lib.match(j) or os.path.exists(j):
+                    elif reg_is_lib.match(j):
+                        libraries += [j]
+                    elif is_abs_path and os.path.exists(j):
                         libraries += [j]
+                    elif mesonlib.is_windows() and not is_abs_path:
```

Comment:
This branch will break if there is random junk (stray link flags, etc.) in one of the properties. Could you please make the check more restrictive, so that only valid libraries are added?

Reply:
How do I know what a valid library is? I don't have a compiler object to hand, since the dependency doesn't intrinsically include a language.

Reply:
I am not sure that there is a good solution. But we should avoid having stuff like `-Dfoobar=stuff.dll` in the link flags. Do you have a link to the CMake documentation on how exactly it is handled there?

If not, then maybe use some sort of regex to check if library name looks OK (starts with an alphanumerical char, etc.)

\<rant\>Who thought it was a good idea to allow plain names like `version` (that are also practically indistinguishable from CMake targets) as valid library names?\</rant\>

Reply:
> I am not sure that there is a good solution. But we should avoid having stuff like -Dfoobar=stuff.dll in the link flags.

OK, so how about the regex as you mentioned, restricting this behaviour to only bare names (`^[a-zA-Z0-9_]+$`)?

> Do you have a link to the CMake documentation on how exactly it is handled there?

It brute-forces across a range of possible prefix/suffix combinations and tries a compiler with them to see which one will work.

> <rant>Who thought it was a good idea to allow plain names like version (that are also practically indistinguishable from CMake targets) as valid library names?</rant>

Yes, I definitely have some opinions on CMake + LLVM + Windows after the past couple of weeks.

Reply:
> OK, so how about the regex as you mentioned, restricting this behaviour to only bare names (^[a-zA-Z0-9_]+$)?

Yes, this regex should work for now. But please also add a comment explaining the situation, otherwise, there will be confusion in the future.

Reply:
Thanks a lot for all the great suggestions. I've implemented them all now and pushed.